grails-app/services/au/org/ala/volunteer/ForumService.groovy

```diff
@@ -5,17 +5,228 @@ import grails.gorm.transactions.Transactional
 import groovy.time.TimeDuration
 import groovy.util.logging.Slf4j
 import org.hibernate.Hibernate
-import org.hibernate.criterion.CriteriaSpecification
+import org.jooq.DSLContext
+import org.jooq.SortOrder
+
+import static org.jooq.impl.DSL.concat
+import static org.jooq.impl.DSL.select
+import static au.org.ala.volunteer.jooq.tables.Project.PROJECT
+import static au.org.ala.volunteer.jooq.tables.Task.TASK
+import static au.org.ala.volunteer.jooq.tables.ForumTopic.FORUM_TOPIC
+import static au.org.ala.volunteer.jooq.tables.ForumMessage.FORUM_MESSAGE
+import static au.org.ala.volunteer.jooq.tables.VpUser.VP_USER
+import static au.org.ala.volunteer.jooq.tables.UserForumWatchList.USER_FORUM_WATCH_LIST
+import static au.org.ala.volunteer.jooq.tables.UserForumWatchListForumTopic.USER_FORUM_WATCH_LIST_FORUM_TOPIC
+import static org.jooq.impl.DSL.count as jCount
```

Comment:
This might be a "me" issue but the space either side of "as" is showing as a black box - does this indicate some unicode style whitespace that isn't an actual space?